File: RecallThatAndroid/app/src/main/java/com/recallthat/Services/search/DefaultSearchService.kt
package com.recallthat.Services.search

import com.recallthat.Models.Memory

class DefaultSearchService : SearchService {
    override fun search(query: String, memories: List<Memory>): List<Memory> {
        if (query.isBlank()) return memories
        return memories.filter { it.searchText.contains(query.trim(), ignoreCase = true) }
    }
}

File: RecallThat/Services/Search/DefaultSearchService.swift
import Foundation

@MainActor
final class DefaultSearchService: SearchServiceProtocol {

    func search(query: String, in memories: [Memory]) async -> [Memory] {
        let trimmed = query.trimmingCharacters(in: .whitespaces)
        guard !trimmed.isEmpty else { return [] }
        let normalized = trimmed.lowercased()
        return memories.filter { $0.searchText.contains(normalized) }
    }
}
